lib/cfg.js

```diff
@@ -0,0 +1,383 @@
+// Copyright (c) 2012-2017, Najjar Chedy
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+
+var _ = require('underscore-node'),
+    logger = require('./logger').logger;
+    
+    
+var InstructionType_jmp = 0;
+var InstructionType_conditionalJmpInst = 1;
+var InstructionType_notRetInst = 2;    
+var InstructionType_retInst = 3;    
+
+
+// this mess is intended to be ported to another file.
+function separateCodeFromData(asmArr) {
+    
+    var labelRegex = /\.L{1}\d+:/;
```

Comment:
Cool: I had to look up the specifics but the `.L` prefix means "local" - https://sourceware.org/binutils/docs/as/Symbol-Names.html -- I don't think it needs to be restricted to a `\d+` though. In practice the compilers generated numbered labels, so this is probably ok, but might be an unnecessary limitation.

Reply:
the name is misleading, I should probably name this jmpLabel regex. I am only interested in getting jmp labels which came in form of: `.L*:` where the start denote a number. there is `.LC*:`, `.LBB3:` and others, that I need to be removed in the process of separating code from data. even though they are not code(assembly instructions) but they are important information that I need to keep.
look at this: https://regex101.com/r/GosSCg/2

Reply:
I'm confused: all the labels after `.L` are potential jump targets, not just those with numbers. Is it important to special case the number ones? That seems to rely heavily on compiler internal names, and I was hoping we don't need to do that. Can you explain ?

Thanks for the regex link, that helped!

Reply:
first, forget to thank you for the link.
when I was analysing gcc output , and when I was using CE for some of my wore, I never crossed a case where a **jmp** instruction used sth other than `.L*` symbol to refer to it's target. This is like I said for gcc, For clang I didn't see sth else than a `.LBB*_*`. please, if you have an example where gcc and clang doesn't use this current labeling. 

I agree that it can be a potential problem for other compilers targeting different architectures because I was considering only output generated by gcc and targetting X86.

Reply:
good point! but this change will be a break to the whole algo. I already thought of a solution to treat the problem independently from compiler but it will require more time to work on.